src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -578,15 +578,34 @@ R"(#include <stdio.h>
         }
     }
 
+    void extract_kinds(const ASR::Cast_t &x, int &arg_kind, int &dest_kind) {
+        dest_kind = ASRUtils::extract_kind_from_ttype_t(x.m_type);
+        ASR::ttype_t *curr_type = ASRUtils::expr_type(x.m_arg);
+        LFORTRAN_ASSERT(curr_type != nullptr)
+        arg_kind = ASRUtils::extract_kind_from_ttype_t(curr_type);
+    }
+
     void visit_Cast(const ASR::Cast_t &x) {
         self().visit_expr(*x.m_arg);
         switch (x.m_kind) {
             case (ASR::cast_kindType::IntegerToReal) : {
-                src = "(float)(" + src + ")";
+                int arg_kind = -1, dest_kind = -1; // arg_kind is not used currently
```

Comment:
Why is `arg_kind` needed? Isn't only the `dest_kind` important in this case?

Reply:
Please, possibly share if it is to be removed. Along with it, it seems we could also remove the function `extract_kinds()`.

Reply:
I just noticed it. It seems it is not unused. We are passing it as a parameter to `extract_kinds()`.

Reply:
Please, possibly share if it is to be removed. Along with it, it seems we could also remove the function `extract_kinds()`. We can get the `dest_kind` directly using `dest_kind = ASRUtils::extract_kind_from_ttype_t(x.m_type);`.

Reply:
> It is mostly present because it seems that there could be a possibility that we might need the `arg_kind` later.

Well, then we will make the change later. :-). Let's use  only`dest_kind` for now.

> Please, possibly share if it is to be removed. Along with it, it seems we could also remove the function `extract_kinds()`. We can get the `dest_kind` directly using `dest_kind = ASRUtils::extract_kind_from_ttype_t(x.m_type);`. (where `x` is of type `const ASR::Cast_t`).

Yes, exactly.

Reply:
Got it.

Reply:
I just added a commit with this suggestion.